### DISCUSSION TITLE
Cygwin vim + nodejs  -> error on start up

### DISCUSSION THREAD

**Discussion by scnew** [neoclide/coc.nvim#4525] [Q&A]
I have the following installed:

$ vim -version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Feb 13 2022 22:15:51)
Garbage after option argument: "-version"
More info with: "vim -h"

$ node -v
v18.14.1

I have configured vim in an attempt to use coc.  I beleive I have installed everything as described in directions, so I don't think I made a simple error. (still possible) :)

When I start vim+coc I get the following errors

    1 [coc.nvim]: node:internal/modules/cjs/loader:1078
    2 [coc.nvim]:   throw err;
    3 [coc.nvim]:   ^
    4 [coc.nvim]: Error: Cannot find module 'C:\home\Ed\.vim\plugged\coc.nvim\build\index.js'
    5 [coc.nvim]:     at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    6 [coc.nvim]:     at Module._load (node:internal/modules/cjs/loader:920:27)
    7 [coc.nvim]:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    8 [coc.nvim]:     at node:internal/main/run_main_module:23:47 {
    9 [coc.nvim]:   code: 'MODULE_NOT_FOUND',
   10 [coc.nvim]:   requireStack: []
   11 [coc.nvim]: }
   12 [coc.nvim]: Node.js v18.14.1
   13 [coc.nvim]: node:internal/modules/cjs/loader:1078
   14 [coc.nvim]:   throw err;
   15 [coc.nvim]:   ^
   16 [coc.nvim]: Error: Cannot find module 'C:\home\Ed\.vim\plugged\coc.nvim\build\index.js'
   17 [coc.nvim]:     at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
   18 [coc.nvim]:     at Module._load (node:internal/modules/cjs/loader:920:27)
   19 [coc.nvim]:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
   20 [coc.nvim]:     at node:internal/main/run_main_module:23:47 {
   21 [coc.nvim]:   code: 'MODULE_NOT_FOUND',
   22 [coc.nvim]:   requireStack: []
   23 [coc.nvim]: }
   24 [coc.nvim]: Node.js v18.14.1
  

Is anyone else using cygwin + vim + Conquer of Completion?

If cygwin cannot be used, what would be a good second choice to Conquer of Completion so that I can get some type of autocompletion?

Many thanks for any response in advance.

**Comment by fannheyward** [comment]
similar issue https://github.com/neoclide/coc.nvim/issues/3094

**Comment by scnew** [comment]
coc.vim - cygwin combination:

There may be some people that get  coc.vim - cygwin working.  The configuration options are not published.  Following all the directions and reading most of the posts about coc.vim - cygwin did not yield a working coc.vim - cygwin for me.  

Version of cygwin:
----------------------------------------------------------------------------------
Cygwin Package Information
Package              Version        Status
cygwin               3.4.10-1       OK
----------------------------------------------------------------------------------

Version of vim:
---------------------------------------------------------------------------------
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Dec 20 2023 06:57:02)
Garbage after option argument: "-version"
More info with: "vim -h"
---------------------------------------------------------------------------------

I installed node.js from https://nodejs.org/en
-----------------------------------------------------------------------------------
]$ node -v
v20.10.0
-----------------------------------------------------------------------------------

Did the following to .bashrc
-----------------------------------------------------------------------------------
alias npm="/cygdrive/c/Program\ Files/nodejs/npm.cmd"
alias node="/cygdrive/c/Program\ Files/nodejs/node.exe"
----------------------------------------------------------------------------------

Made links
---------------------------------------------------------------------------------
using cmd.exe
mklink /j "c:/home" "c:/cygwin64/home"
---------------------------------------------------------------------------------

Started vim.  I use plug install to install coc.vim.
Used the :PlugInstall to install coc.vim
Sucessfully installed coc.vim.

Restarted.  I got nothing for coc.vim.

Debug shows the following:

Executing CursorMoved Autocommands for "*"
 autocommand call s:Highlight_Matching_Pair()
 
   [coc.nvim]: node:internal/modules/cjs/loader:1147
  [coc.nvim]:   throw err;
 [coc.nvim]:   ^
  [coc.nvim]: Error: Cannot find module 'C:\home\Ed\.dotfiles\vim\.vim\plugged\coc.nvim\build\ind
  [coc.nvim]:     at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
  [coc.nvim]:     at Module._load (node:internal/modules/cjs/loader:985:27)
  [coc.nvim]:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:
  [coc.nvim]:     at node:internal/main/run_main_module:28:49 {
  [coc.nvim]:   code: 'MODULE_NOT_FOUND',
   [coc.nvim]:   requireStack: []
   [coc.nvim]: }
  [coc.nvim]: Node.js v20.10.0
   client coc abnormal exit with: 1
   Executing ModeChanged Autocommands for "*"
   autocommand :call airline#update_tabline()
  
There may be good info on coc.vim - cygwin installation.. I have not found it.  What magic have other used to get this working.. is a mystery that I have not seen posted to date. -
